django/src/rdwatch_scoring/views/vector_tile.py

```diff
@@ -0,0 +1,234 @@
+
+from rdwatch.db.functions import ExtractEpoch, GroupExcludeRowRange
+from rdwatch_scoring.models import EvaluationRun, Observation, Region, Site
+
+from .model_run import router
+
+
+def _get_vector_tile_cache_key(
+    evaluation_run_uuid: UUID4, z: int, x: int, y: int, timestamp: datetime
+) -> str:
+    return '|'.join(
+        [
+            Site.__name__,
+            str(evaluation_run_uuid),
+            'vector-tile',
+            str(z),
+            str(x),
+            str(y),
+            str(timestamp),
+        ]
+    ).replace(' ', '_')
+
+
+@router.get('/{evaluation_run_uuid}/vector-tile/{z}/{x}/{y}.pbf/')
+def vector_tile(
+    request: HttpRequest, evaluation_run_uuid: UUID4, z: int, x: int, y: int
+):
+    evaluation_run = get_object_or_404(EvaluationRun, pk=evaluation_run_uuid)
+
+    latest_timestamp = evaluation_run.start_datetime
+
+    # Generate a unique cache key based on the model run ID, vector tile coordinates,
+    # and the latest timestamp
+    cache_key = _get_vector_tile_cache_key(
+        evaluation_run_uuid, z, x, y, latest_timestamp
+    )
+
+    tile = cache.get(cache_key)
+
+    # Generate the vector tiles and cache them if there's no hit
+    if tile is None:
+        envelope = Func(z, x, y, function='ST_TileEnvelope')
+        intersects = Q(
+            Func(
+                'transformedgeom',
+                envelope,
+                function='ST_Intersects',
+                output_field=BooleanField(),
+            )
+        )
+        transform = Func(
+            'geomfromtext', 3857, function='ST_Transform', output_field=GeometryField()
+        )
+        mvtgeom = Func(
+            'transformedgeom',
+            envelope,
+            function='ST_AsMVTGeom',
+            output_field=Field(),
+        )
+        geomfromuniongeometrytext = Func(
+            'union_geometry', 4326, function='ST_GeomFromText', output_field=Field()
+        )
+        geomfromobservationtext = Func(
+            'geometry', 4326, function='ST_GeomFromText', output_field=Field()
+        )
+
+        site_queryset = (
+            Site.objects.filter(evaluation_run_uuid=evaluation_run_uuid)
+            .alias(geomfromtext=geomfromuniongeometrytext)
+            .alias(transformedgeom=transform)
+            .filter(intersects)
+            .values()
+            .annotate(
+                id=F('site_id'),
+                mvtgeom=mvtgeom,
+                configuration_id=F('evaluation_run_uuid'),
+                label=Case(
+                    When(
+                        Q(status_annotated__isnull=False),
+                        Lower(Replace('status_annotated', Value(' '), Value('_'))),
+                    ),
+                    default=Value('unknown'),
+                ),  # This needs a version to be scoring coloring,
+                # but that needs some coordination with kitware
+                timestamp=ExtractEpoch('evaluation_run_uuid__start_datetime'),
```

Comment:
```suggestion
                timestamp=ExtractEpoch('start_date'),
```
This may be removed as well in our vector-tile endpoint I don't know if it's used but it is a timestamp which I think is a created timestamp.  It has no relation to any data I believe in the Evaluation.  Please double check this.

Reply:
Yeah, it's a created timestamp, this makes sense to me.